static/js/message_view_header.js

```diff
@@ -146,13 +114,15 @@ function build_message_view_header(filter) {
 // the searchbar has the right text.
 export function reset_searchbox_text() {
     let search_string = narrow_state.search_string();
-    if (search_string !== "") {
-        if (!page_params.search_pills_enabled && !narrow_state.filter().is_search()) {
-            // saves the user a keystroke for quick searches
-            search_string = search_string + " ";
-        }
-        $("#search_query").val(search_string);
+    if (
+        search_string !== "" &&
+        !page_params.search_pills_enabled &&
+        !narrow_state.filter().is_search()
+    ) {
+        // saves the user a keystroke for quick searches
+        search_string = search_string + " ";
     }
+    $("#search_query").val(search_string);
```

Comment:
we now always want to reset, even with the empty string, since the searchbar is always visible (sometimes collapsed and sometimes expanded)